packages/dd-trace/src/exporters/common/request.js

```diff
@@ -93,15 +95,29 @@ function request (data, options, callback) {
   options.agent = isSecure ? httpsAgent : httpAgent
 
   const onResponse = res => {
-    let responseData = ''
+    const chunks = []
 
     res.setTimeout(timeout)
 
-    res.on('data', chunk => { responseData += chunk })
+    res.on('data', chunk => {
+      const bufferChunk = Buffer.isBuffer(chunk) ? chunk : Buffer.from(chunk)
+      chunks.push(bufferChunk)
+    })
     res.on('end', () => {
+      const buffer = Buffer.concat(chunks)
       activeRequests--
 
       if (res.statusCode >= 200 && res.statusCode <= 299) {
+        let responseData = ''
+        if (res.headers['content-encoding'] === 'gzip') {
+          try {
+            responseData = zlib.gunzipSync(buffer).toString()
```

Comment:
Please do not use the sync version. This will have a huge latency hit. I know this will only run in CI currently, but someone could use it in the future and not realize the latency consequences.

Reply:
changed to async in https://github.com/DataDog/dd-trace-js/pull/3992/commits/900888df73a32a65bcb87974b7c0095626267b69